src/chainedvector.jl

```diff
@@ -629,6 +629,8 @@ Base.@propagate_inbounds function Base.insert!(A::ChainedVector{T, AT}, i::Integ
     return A
 end
 
+Base.vcat(A::ChainedVector{T, AT}) where {T, AT <: AbstractVector{T}} = A
+
 function Base.vcat(A::ChainedVector{T, AT}, arrays::ChainedVector{T, AT}...) where {T, AT <: AbstractVector{T}}
```

Comment:
I'd suggest folding this into the same method, either with a fast path like
```suggestion
function Base.vcat(A::ChainedVector{T, AT}, arrays::ChainedVector{T, AT}...) where {T, AT <: AbstractVector{T}}
    isempty(arrays) && return Base.unaliascopy(A)
```
or by fixing the source of the error directly by passing `init=0` to `sum` a few lines down. (The latter would require bumping the minimum Julia compat to at least 1.6, but IMO that's fine.)